lightning/src/chain/mod.rs

```diff
@@ -86,8 +86,11 @@ pub trait Listen {
 
 	/// Notifies the listener that one or more blocks were removed in anticipation of a reorg.
 	///
-	/// Indicates the new best tip is the provided [`BestBlock`].
-	fn blocks_disconnected(&self, new_best_block: BestBlock);
+	/// The provided [`BestBlock`] is the new best block after disconnecting blocks in the reorg
+	/// but before connecting new ones (i.e. the "fork point" block). For backwards compatibility,
+	/// you may instead walk the chain backwards, calling `blocks_disconnected` for each block
+	/// which is disconnected in a reorg.
+	fn blocks_disconnected(&self, fork_point_block: BestBlock);
```

Comment:
To avoid confusion, maybe we should rename this `fork_detected` then? Though I guess the naming wouldn't be intuitive for the backwards compatibility case. But maybe that is not a concern given `lightning-block-sync` doesn't support backwards compatibility. When would someone need this? When they are using their own implementation of `Listen` and not using `lightning-block-sync`?